src/adapted/mod.rs

```diff
@@ -257,8 +258,19 @@ impl<'a> CustomTryFrom<raw::client_messages::WriteEventsCompleted<'a>> for Adapt
                     //  * separate (instead of newtype for tuple)
                     // these must be options, as for idempotent writes the positions might not be
                     // returned. both seem to be returned always.
-                    prepare_position: msg.prepare_position.map(|x| x.into()),
-                    commit_position: msg.commit_position.map(|x| x.into()),
+
+                    // FIXME once this sort of error return were removed,
+                    // it will be possible to use the Carrier syntax (?)
```

Comment:
I think the macros are the only nice way to go here. If these methods did not return the argument as part of the `Err` we'd have to clone those types which are quite big and can have 16MB of data associated.